ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -169,31 +208,17 @@ def install_dcos(config, block=False, state_json_dir=None, hosts=None, async_del
         hosts = []
     assert isinstance(hosts, list)
 
-    # Role specific parameters
-    role_params = {
-        'master': {
-            'tags': {'role': 'master', 'dcos_install_param': 'master'},
-            'hosts': config['master_list']
-        },
-        'agent': {
-            'tags': {'role': 'agent', 'dcos_install_param': 'slave'},
-            'hosts': config.get('agent_list', [])
-        },
-        'public_agent': {
-            'tags': {'role': 'public_agent', 'dcos_install_param': 'slave_public'},
-            'hosts': config.get('public_agent_list', [])
-        }
-    }
-
     bootstrap_tarball = _get_bootstrap_tarball()
     log.debug("Local bootstrap found: %s", bootstrap_tarball)
 
     targets = []
     if hosts:
         targets = hosts
     else:
-        for role, params in role_params.items():
-            targets += [Node(node, params['tags']) for node in params['hosts']]
+        if state_json_dir is None:
```

Comment:
yes, you are reading it right. check `state_json_dir` if the var is not None (this will happen if we use JSONDelegate in web mode) then search for nodes there. Otherwise get them from a config. This is definitely implicit, i will fix it.